app/javascript/mastodon/features/ui/util/fullscreen.ts

```diff
@@ -0,0 +1,66 @@
+/* eslint-disable @typescript-eslint/no-floating-promises */
+/* eslint-disable @typescript-eslint/prefer-nullish-coalescing */
+// APIs for normalizing fullscreen operations. Note that Edge uses
+// the WebKit-prefixed APIs currently (as of Edge 16).
```

Comment:
Explaining the shenanigans in this file 😄: TypeScript's types these days assume you're in an evergreen/relatively-recently-supported browser. All of those browsers support `document.fullscreenElement` and other APIs that this file seems to assume might not be supported. The type additions in this file are telling TypeScript _not_ to make that assumption.

Alternately, a cleaner solution would be to not keep the older browser detection in place. That'd remove the need for this TypeScript type stuffs too.

Reply:
I think we can drop browser detection as we are moving to supporting modern browsers only.

According to caniuse, we still need to check for the webkit prefixed methods,  but we can probably drop the others.